pkg/stats/collector.go

```diff
@@ -173,20 +171,18 @@ func NewBufferedCollector(installationID string, cfg Config, opts ...BufferedCol
 	}
 	s := &BufferedCollector{
 		cache:           make(keyIndex),
-		writes:          make(chan Metric, collectorEventBufferSize),
+		updatesCh:       make(chan Metric, collectorEventBufferSize),
 		runtimeStats:    map[string]string{},
 		flushTicker:     &TimeTicker{ticker: time.NewTicker(cfg.FlushInterval)},
 		flushSize:       cfg.FlushSize,
 		heartbeatTicker: &TimeTicker{ticker: time.NewTicker(heartbeatInterval)},
 		installationID:  installationID,
 		processID:       uuid.Must(uuid.NewUUID()).String(),
-		pendingWrites:   sync.WaitGroup{},
-		pendingRequests: sync.WaitGroup{},
-		ctxCancelled:    0,
-		done:            make(chan bool),
-		sendTimeout:     sendTimeout,
+		inFlight:        sync.WaitGroup{},
+		closed:          0,
 		extended:        cfg.Extended,
 		log:             logging.Default(),
+		sendCh:          make(chan *InputEvent, 1), // buffered as like to check if sender is free
```

Comment:
Didn't understand your comment.

Reply:
single item channel to handle one post at a time